internal/v1/handler.go

```diff
@@ -86,17 +88,25 @@ func (h *Handlers) GetDistributions(ctx echo.Context) error {
 	dr := h.server.distroRegistry(ctx)
 	userID, err := h.server.getIdentity(ctx)
 	if err != nil {
-		return err
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to get user identity: %v", err))
 	}
 
+	distributions, err := h.filterDistributions(userID.OrgID(), *dr)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to filter distributions").SetInternal(err)
```

Comment:
nit: just to give user extra context why we were filtering the distributions :)
```suggestion
		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to filter available distributions").SetInternal(err)
```